raster/apply-mosaic-rule-to-rasters/src/main/java/com/esri/samples/apply_mosaic_rule_to_rasters/ApplyMosaicRuleToRastersSample.java

```diff
@@ -0,0 +1,214 @@
+
+          // enable UI interaction once the raster layer has loaded
+          controlsVBox.setVisible(true);
+
+          // create a mosaic rule for the image service raster if one does not already exist
+          if (imageServiceRaster.getMosaicRule() == null) {
+            mosaicRule = new MosaicRule();
+            imageServiceRaster.setMosaicRule(mosaicRule);
+          }
+        } else {
+          // show alert if raster layer fails to load.
+          new Alert(Alert.AlertType.ERROR, "Error loading raster layer.").show();
+        }
+      });
+
+      // set the mosaic rule of the image service raster based on rule chosen from the combo box
+      mosaicRuleComboBox.getSelectionModel().selectedItemProperty().addListener(e -> {
+        // create a new mosaic rule
+        mosaicRule = new MosaicRule();
+
+        switch (mosaicRuleComboBox.getSelectionModel().getSelectedItem()) {
+          case "Default":
+            mosaicRule.setMosaicMethod(MosaicMethod.NONE);
+            break;
+          case "Northwest":
+            mosaicRule.setMosaicMethod(MosaicMethod.NORTHWEST);
+            mosaicRule.setMosaicOperation(MosaicOperation.FIRST);
+            break;
+          case "Center":
+            mosaicRule.setMosaicMethod(MosaicMethod.CENTER);
+            mosaicRule.setMosaicOperation(MosaicOperation.BLEND);
+            break;
+          case "By attribute":
+            mosaicRule.setMosaicMethod(MosaicMethod.ATTRIBUTE);
+            mosaicRule.setSortField("OBJECTID");
+            break;
+          case "Lock raster":
+            mosaicRule.setMosaicMethod(MosaicMethod.LOCK_RASTER);
+            mosaicRule.getLockRasterIds().clear();
+            mosaicRule.getLockRasterIds().addAll(Arrays.asList(1L, 7L, 12L));
+            break;
+        }
+        // set the mosaic rule of the image service raster
+        imageServiceRaster.setMosaicRule(mosaicRule);
+      });
+
+      // add the map view, control panel and progress indicator to the stack pane
+      stackPane.getChildren().addAll(mapView,controlsVBox, progressIndicator);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Creates a UI with a drop down box.
+   */
+  private void setupUI() {
+    // create a label
+    Label mosaicRuleLabel = new Label("Choose a mosaic rule: ");
+    mosaicRuleLabel.setTextFill(Color.WHITE);
+
+    // create a combo box
+    mosaicRuleComboBox = new ComboBox<>();
+    mosaicRuleComboBox.setMaxWidth(Double.MAX_VALUE);
+
+    // add the mosaic rules to the combo box
+    mosaicRuleComboBox.getItems().addAll("Default", "Northwest", "Center", "By attribute", "Lock raster");
+
+    // set the default combo box value
+    mosaicRuleComboBox.getSelectionModel().select(0);
```

Comment:
If you wanted, you could instantiate the combobox directly with the items, instead of adding them in a separate line,
```suggestion
    // create a combo box with the mosaic rules
    mosaicRuleComboBox = new ComboBox<>(
            FXCollections.observableArrayList("Default", "Northwest", "Center", "By attribute", "Lock raster"));
    mosaicRuleComboBox.setMaxWidth(Double.MAX_VALUE);

    // set the default combo box value
    mosaicRuleComboBox.getSelectionModel().select(0);

```